tsrc/workspace.py

```diff
@@ -321,15 +336,31 @@ def process(self, repo):
             self.check_branch(repo, repo_path)
             self.sync_repo_to_branch(repo_path)
 
+    def get_default_branch(self, repo_path):
+        _, remote_branches = tsrc.git.run_git(repo_path, "branch", "--remotes", "--points-at", "refs/remotes/origin/HEAD", raises=False)
```

Comment:
maybe there's a better way than paring the output of `git branch`.

Reply:
Also, I *think* that if `production` is the default branch, running `git clone` without arguments should do the trick.

Reply:
I agree, that's what I tried first... But then all tests are failing cause `check_branch`

Reply:
> But then all tests are failing cause check_branch

Then we should fix the tests :stuck_out_tongue:  We can help if you have a test failure with a backtrace to share :)

Reply:
ok, earlier I could not remember why I needed this one... So the problem is in _ensure_git_state in the case `self.clone_path.exists()` you need the default branch name for your repository in order to do `git checkout -B`...
I have not found anything better yet for this one, but any idea would be welcome :)

Reply:
> _ensure_git_state in the case self.clone_path.exists() you need the default branch name for your repository in order to do git checkout -B...

This should not be the case. We should read the branch from the config file in `<workspace/.tsrc/manifest.yml>` I think.